jiraServerPlugin/src/main/java/com/opsgenie/plugin/service/OpsgeniePluginSettingsManagerImpl.java

```diff
@@ -102,36 +116,38 @@ public void createOpsgenieUser() throws OpsgenieUserCreationFailedException {
 
     @Override
     public void saveSettings(OpsgeniePluginSettings opsgeniePluginSettings) {
+        logger.debug("Saving Settings...");
         if (opsgeniePluginSettings == null) {
-            throw new ValidationException("OpsgeniePluginSettings cannot be null");
+            throw new ValidationException("Opsgenie Plugin Settings cannot be null");
         }
         opsgeniePluginSettings.validateBeforeSave();
         pluginSettings.put(OG_PLUGIN_SETTINGS, gson.toJson(opsgeniePluginSettings));
+        logger.info("Settings saved successfully!");
     }
 
     @Override
-    public void updateSettings(OpsgeniePluginSettings opsgeniePluginSettings) {
+    public void updateSettings(OpsgeniePluginSettings opsgeniePluginSettings) throws OpsgenieUserCreationFailedException {
+        logger.debug("Settings already exist. Updating...");
         ConnectionUpdateDto connectionUpdateDto = new ConnectionUpdateDto();
-        connectionUpdateDto.setServerId(getServerId().orElseThrow(() -> new ValidationException("serverId is empty!")));
+        connectionUpdateDto.setServerId(getServerId().orElseThrow(() -> new ValidationException("Could not find server Id!")));
         connectionUpdateDto.setServerUrl(opsgeniePluginSettings.getServerUrl());
         OpsgeniePluginSettings existingSettings = getSettings().orElseThrow(() -> new ValidationException("There is no configuration to update!"));
         String endpoint = SETUP_ENDPOINT;
         if (!StringUtils.equals(opsgeniePluginSettings.getApiKey(), existingSettings.getApiKey())) {
             endpoint = SETUP_ENDPOINT + "/owner";
         }
         SendResult result = opsgenieClient.put(opsgeniePluginSettings.getBaseUrl() + endpoint, opsgeniePluginSettings.getApiKey(), gson.toJson(connectionUpdateDto));
-        if (!result.isSuccess()) {
-            throw new ValidationException("Could not update the plugin settings. Reason: " + result.getFailReason());
-        } else {
+
+        if (result.isSuccess()) {
+            if (!findUser().isPresent()) {
+                createUser(existingSettings);
```

Comment:
We can create / update the existing endpoint to accept the user password as an argument. So that we can keep it alway in sync whenever a new user is created.